k8s/security_context.go

```diff
@@ -1,16 +1,34 @@
 package k8s
 
 import (
+	"strconv"
+
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-func defaultSecurityContext() *corev1.SecurityContext {
+func defaultSecurityContext(client *kubernetes.Clientset) *corev1.SecurityContext {
 	runAsNonRoot := true
-	return &corev1.SecurityContext{
+	sc := &corev1.SecurityContext{
 		Privileged:               new(bool),
 		AllowPrivilegeEscalation: new(bool),
 		RunAsNonRoot:             &runAsNonRoot,
 		Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}},
 		SeccompProfile:           nil,
 	}
+	if info, err := client.ServerVersion(); err == nil {
+		var maj, min int64
+		maj, err = strconv.ParseInt(info.Major, 10, 64)
+		if err != nil {
+			maj = -1
+		}
+		min, err = strconv.ParseInt(info.Minor, 10, 64)
+		if err != nil {
+			min = -1
+		}
+		if maj >= 1 && min >= 24 {
```

Comment:
Who knows if there ever will be a kubernetes 2.0, but if so, we'll need to change this logic a little bit. :)